src/status_im/contexts/onboarding/events.cljs

```diff
@@ -1,31 +1,192 @@
 (ns status-im.contexts.onboarding.events
   (:require
     [quo.foundations.colors :as colors]
+    [re-frame.interceptor :as interceptor]
+    [react-native.mmkv :as mmkv]
     status-im.common.biometric.events
     [status-im.constants :as constants]
     [status-im.contexts.shell.constants :as shell.constants]
+    [status-im.feature-flags :as ff]
     [taoensso.timbre :as log]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]
     [utils.security.core :as security]))
 
+(defn notifications-setup-start
+  [{:keys [db]}
+   [{:keys [enable-notifications?
+            enable-news-notifications?
+            biometrics?
+            onboarding?
+            syncing?]}]]
+  {:db (if-not onboarding?
+         db
+         (update-in db
+                    [:onboarding/profile]
+                    assoc
+                    :enable-notifications?      enable-notifications?
+                    :enable-news-notifications? enable-news-notifications?))
+   :fx (cond-> []
+         (and (not onboarding?) enable-notifications? enable-news-notifications?)
+         (conj [:dispatch [:notifications/news-notifications-switch true]])
+
+         (and (not onboarding?) enable-notifications?)
+         (conj [:dispatch [:push-notifications/switch true]])
```

Comment:
If we're not inside the onboarding flow, for example inside the login flow, then we want to activate the push notifications immediately. Though during onboarding we'll defer activating them until we've completed the onboarding process (creating a profile).